tests/unit_tests/test_internal_api.py

```diff
@@ -81,6 +81,13 @@ def test_download_write_file_fetches_iff_file_checksum_mismatched(
             assert response is None
 
 
+def test_internal_api_with_no_write_global_config_dir():
+    tempdir = tempfile.mkdtemp()
+    with patch.dict("os.environ", WANDB_CONFIG_DIR=tempdir):
+        os.chmod(tempdir, 0o444)
+        internal.InternalApi()
```

Comment:
Pytest has a built-in `tmp_path` fixture, so you can rewrite this as
```suggestion
def test_internal_api_with_no_write_global_config_dir(tmp_path):
    with patch.dict("os.environ", WANDB_CONFIG_DIR=tmp_path):
        os.chmod(tmp_path, 0o444)
        internal.InternalApi()
```